Signal/src/ViewControllers/DebugUI/DebugUIMessages.m

```diff
@@ -1980,7 +1983,10 @@ + (DebugUIMessagesAction *)fakeRandomTextAction:(TSThread *)thread
                 OWSAssert(messageToQuote);
                 DDLogVerbose(@"%@ %@", self.logTag, label);
                 [DDLog flushLog];
-                ConversationViewItem *viewItem = [[ConversationViewItem alloc] initWithInteraction:messageToQuote isGroupThread:thread.isGroupThread transaction:transaction];
+                ConversationViewItem *viewItem = [[ConversationViewItem alloc] initWithInteraction:messageToQuote
+                                                                                     isGroupThread:thread.isGroupThread
+                                                                                       transaction:transaction
+                                                                                        layoutInfo:layoutInfo];
```

Comment:
I'm somewhat resistant to caching the layoutInfo on the ConversationViewItem. As you can see here, we've perviously used them in multiple contexts, often outside of the ConversationViewController.

Also, it seems kind of awkward in the MessageDetailViewController.

If possible, it seems like the ConversationViewLayout or the ConversationViewController would be a better place to maintain a cache of any layout attributes.

Reply:
* As an aside, I'm thinking of renaming `ConversationLayoutInfo` to `ConversationAppearance` or `ConversationStyle`, since I think it'll better capture the intention. 
* Conversation cell layout and styling (e.g. colors) is contextual (to the thread, the containing view size, etc.) so we definitely need either this kind of class or possibly a protocol/delegate pattern.
* `ConversationViewController` is a massive file - 5k+ lines.  I'm hoping that we can slowly decompose it into smaller components with well-defined responsibilities.  And I'm trying to resist it growing any larger.  The "conversation style" info seems like an opportunity to gather/bundle some related code.
* `ConversationViewLayout` is a layout for Conversation view's collection view - none of its logic or state is relevant to message details view.  Even if we did move these properties and logic there, we'd need to do just as much work to keep it's "view width" state updated.
* There are other alternatives that would make using this "appearance/style" class a little easier, but they would require KVO of the root view's frame & bounds, and the docs discourage that - and the hacks to do it are awkward and not future-proof.

Want to discuss when you come online?